dash_labs/plugins/pages.py

```diff
@@ -294,6 +297,40 @@ def _import_layouts_from_pages():
                 )
 
 
+def _register_page_from_markdown_file(filename, app):
+    """
+    Extracts and runs dash.register_page() from the "front matter" of a markdown file in the pages folder.
+    Front matter is yaml defined with three dashes:
+    ---
+    register_page:
+      path: "/home"
+      order: 2
+    MarkdownAIO:
+      exec: True
+      side_by_side: True
+    ---
+    """
+    import frontmatter
```

Comment:
I was curious to see how this library parsed YAML, given my `exec` / `eval` concerns when writing the code fence parser. 

It looks like frontmatter depends on PyYAML's safe mode: https://github.com/eyeseast/python-frontmatter/blob/1c49958fdd504691fc842045425ea298316aa643/frontmatter/default_handlers.py#L224-L228

So then looking into PyYAML, I see some stuff like: 
> Note that the ability to construct an arbitrary Python object may be dangerous if you receive a YAML document from an untrusted source such as the Internet. The function yaml.safe_load limits this ability to simple Python objects like integers or lists.

from https://pyyaml.org/wiki/PyYAMLDocumentation

Then googled around `yaml.safe_load` and came across https://github.com/yaml/pyyaml/issues/420

They fixed the issue which is great but the fact that an issue happened in the first place makes me feel a little concerned that there could be more issues lurking in that library.

So I might consider supporting a smaller subset of the YAML library and doing our own parsing with JSON, similar to what we did with the code fences.

I'll keep thinking a bit about this